assets/js/components/user-input/UserInputKeywords.js

```diff
@@ -85,30 +86,59 @@ export default function UserInputKeywords( { slug, max } ) {
 		] );
 	}, dependencies );
 
-	const onKeyDown = useCallback( ( index, { keyCode } ) => {
+	const onKeyDown = useCallback( ( index, { keyCode, target } ) => {
 		const nonEmptyValues = values.filter( ( value ) => value.length > 0 );
+		const nonEmptyValuesLength = nonEmptyValues.length;
+
 		if ( ( keyCode === ENTER || keyCode === COMMA ) && nonEmptyValues.length < max ) {
 			updateKeywords( [
 				...values.slice( 0, index + 1 ),
 				'',
 				...values.slice( index + 1 ),
 			] );
 
-			setTimeout( () => {
-				const input = keywordsContainer.current.querySelector( `#${ slug }-keyword-${ index + 1 }` );
-				if ( input ) {
-					input.focus();
-				}
-			}, 50 );
+			setCanDeleteKeyword( true ); // New keyword has been added, so hitting backspace now will remove it
+			setTimeout( () =>
+				focusInput( `#${ slug }-keyword-${ index + 1 }` )
+			, 50 );
```

Comment:
This code style is a bit awkward; do we _need_ to be returning `focusInput()` here? I don't think the return value here is needed, so seems like this is using this syntax to save on curly braces but not because it needs to return.

Let's be clearer with:
```suggestion
			setTimeout( () => {
				focusInput( `#${ slug }-keyword-${ index + 1 }` );
			}, 50 );
```

Sure, it's more characters but I think expresses the intent better. Whenever I see an implicit return I assume the return value matters—because it doesn't here let's avoid it.

Same goes for other bits of this PR 🙂